src/libexpr/primops/fetchTree.cc

```diff
@@ -36,6 +36,9 @@ void emitTreeAttrs(
         mkString(*state.allocAttr(v, state.symbols.create("shortRev")), rev->gitShortRev());
     }
 
+    if (input.getType() == "git")
+        mkBool(*state.allocAttr(v, state.symbols.create("submodules")), maybeGetBoolAttr(input.attrs, "submodules").value_or(false));
```

Comment:
This should only be done for `fetchGit`.

Reply:
Can you explain to me why this shouldn't happen for e.g. `fetchTree { type = "git"; }`?

Reply:
We're trying to avoid returning unnecessary information from `fetchTree` since that might influence the evaluation result. But it probably doesn't matter much here.